cpp/daal/src/algorithms/dtrees/dtrees_train_data_helper.i

```diff
@@ -189,36 +204,109 @@ public:
     size_t size() const { return _aResponse.size(); }
     TResponse response(size_t i) const { return _aResponse[i].val; }
     const Response * responses() const { return _aResponse.get(); }
+
     bool reset(size_t n)
     {
         _aResponse.reset(n);
         return _aResponse.get() != nullptr;
     }
 
-    virtual bool init(const NumericTable * data, const NumericTable * resp, const IndexType * aSample)
+    bool resetWeight(size_t n)
     {
-        super::init(data, resp);
+        _aWeight.reset(n);
+        return _aWeight.get() != nullptr;
+    }
+
+    virtual bool init(const NumericTable * data, const NumericTable * resp, const IndexType * aSample, const NumericTable * weight = nullptr)
+    {
+        super::init(data, resp, weight);
         DAAL_ASSERT(_aResponse.size());
-        if (aSample)
+        if (_aWeight.get()) // use weight
```

Comment:
Unclear comment: does it means use 'weight' parameter instead of field _aWeight? I think you wanted to comment true branch instead:
if (cond)
{
  // comment 1
}
else
{
  // comment 2
}

Reply:
It was fixed